gremlin-go/driver/connection_test.go

```diff
@@ -1,18 +1,117 @@
+specific language governing permissions and limitations
+under the License.
+*/
+
 package gremlingo
 
 import (
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/text/language"
 )
 
 const runIntegration = true
+const testHost string = "localhost"
+const testPort int = 8182
 
 func TestConnection(t *testing.T) {
+	t.Run("Test g.V().count()", func(t *testing.T) {
+		if runIntegration {
+			remote, err := NewDriverRemoteConnection(testHost, testPort)
+			assert.Nil(t, err)
+			assert.NotNil(t, remote)
+			g := _Traversal().WithRemote(remote)
+
+			// Drop vertices that were added.
+			_, err = g.V().Drop().Iterate()
+			assert.Nil(t, err)
+
+			// Read count again, should be 0.
+			results, err := g.V().Count().ToList()
+			assert.Nil(t, err)
+			assert.NotNil(t, results)
+			assert.Equal(t, 1, len(results))
+			var count int32
+			count, err = results[0].GetInt32()
+			assert.Nil(t, err)
+			assert.Equal(t, int32(0), count)
+			time.Sleep(50 * time.Millisecond)
+
+			// Read count again, should be 0, same as above.
+			results, err = g.V().HasLabel("Test").Count().ToList()
+			assert.Nil(t, err)
+			assert.NotNil(t, results)
+			assert.Equal(t, 1, len(results))
+			count, err = results[0].GetInt32()
+			assert.Nil(t, err)
+			assert.Equal(t, int32(0), count)
+			time.Sleep(50 * time.Millisecond)
+
+			// Add 5 vertices.
+			_, err = g.
+				AddV("person").Property("name", "Lyndon").
+				AddV("person").Property("name", "Simon").
+				AddV("person").Property("name", "Yang").
+				AddV("person").Property("name", "Rithin").
+				AddV("person").Property("name", "Alexey").
+				Iterate()
+			assert.Nil(t, err)
+			time.Sleep(50 * time.Millisecond)
+
+			// Read count again, should be 5.
+			results, err = g.V().Count().ToList()
+			assert.Nil(t, err)
+			assert.NotNil(t, results)
+			assert.Equal(t, 1, len(results))
+			count, err = results[0].GetInt32()
+			assert.Nil(t, err)
+			assert.Equal(t, int32(5), count)
+			time.Sleep(50 * time.Millisecond)
+
+			// Drop vertices that were added.
+			_, err = g.V().Drop().Iterate()
+			assert.Nil(t, err)
+			time.Sleep(50 * time.Millisecond)
+		}
+	})
+
+	t.Run("Test addV()", func(t *testing.T) {
+		if runIntegration {
+			remote, err := NewDriverRemoteConnection(testHost, testPort)
+			g := _Traversal().WithRemote(remote)
+
+			// Add 5 vertices.
+			_, err = g.
+				AddV("person").Property("name", "Lyndon").
+				AddV("person").Property("name", "Simon").
+				AddV("person").Property("name", "Yang").
+				AddV("person").Property("name", "Rithin").
+				AddV("person").Property("name", "Alexey").
+				Iterate()
+			assert.Nil(t, err)
+		}
+	})
+
 	t.Run("Test createConnection", func(t *testing.T) {
 		if runIntegration {
-			connection, err := createConnection("localhost", 8181, newLogHandler(&defaultLogger{}, Info, language.English))
+			connection, err := createConnection(testHost, testPort, newLogHandler(&defaultLogger{}, Info, language.English))
```

Comment:
How createConnection is expected to be used by an app (outside of the tests) - newLogHandler is an internal function, are we going to provide a default one? Can it be optional, so the default logger is provided?

Reply:
createConnection is package private (hence lower case name). External users should use DriverRemoteConnection or Client instead, which handle logger creation internally.